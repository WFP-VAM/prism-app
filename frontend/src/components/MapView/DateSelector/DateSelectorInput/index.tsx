import React, { forwardRef, Ref } from 'react';
import { Button } from '@material-ui/core';

interface DateSelectorInputProps {
  value?: string;
  onClick?: () => void;
}

const DateSelectorInput = forwardRef(
  (
    { value, onClick }: DateSelectorInputProps,
    ref?: Ref<HTMLButtonElement>,
  ) => {
    return (
      <Button variant="outlined" onClick={onClick} ref={ref}>
        {value}
      </Button>
    );
  },
);

export default DateSelectorInput;
