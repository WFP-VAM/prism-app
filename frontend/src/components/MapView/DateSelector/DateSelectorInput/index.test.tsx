import { render } from '@testing-library/react';
import React from 'react';
import DateSelectorInput from '.';

test('renders as expected', () => {
  const { container } = render(
    <DateSelectorInput value="Some Value" onClick={() => {}} />,
  );
  expect(container).toMatchSnapshot();
});
