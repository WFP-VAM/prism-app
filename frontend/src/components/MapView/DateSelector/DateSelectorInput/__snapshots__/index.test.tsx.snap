// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <mock-button
    variant="outlined"
  >
    Some Value
  </mock-button>
</div>
`;
